app/models/voxelytics/ElasticsearchClient.scala

```diff
@@ -0,0 +1,95 @@
+package models.voxelytics
+
+import com.scalableminds.util.mvc.MimeTypes
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.webknossos.datastore.rpc.RPC
+import com.typesafe.scalalogging.LazyLogging
+import models.voxelytics.VoxelyticsLogLevel.VoxelyticsLogLevel
+import play.api.http.HeaderNames
+import play.api.libs.json.{JsArray, JsNumber, JsValue, Json}
+import utils.WkConf
+
+import java.util.UUID
+import javax.inject.Inject
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.ExecutionContext
+
+class ElasticsearchClient @Inject()(wkConf: WkConf, rpc: RPC)(implicit ec: ExecutionContext)
+    extends LazyLogging
+    with MimeTypes {
+
+  private lazy val conf = wkConf.Voxelytics.Elasticsearch
+  val SCROLL_SIZE = 10000
+
+  private def pollUntilServerIsAvailable: Fox[Unit] =
+    // TODO: Implement polling with conf.startupTimeout
+    Fox.successful()
```

Comment:
@fm3 I need some help here. In JS I implemented a function that waits until the elasticsearch server is available and initializes the index (if necessary). The function result is memoized, so all callers get the same "promise". A promise is probably the same as a future in Scala. Now, I don't know how to implement `setTimeout` in Scala. I think you mentioned, I need an actor system for that? https://github.com/scalableminds/voxelytics/blob/master/voxelytics/workflows/workflow-viewer/server/logging.js#L10-L62

Reply:
I think I figured it out now. Not sure what the `lifecycle` does, though. https://github.com/scalableminds/webknossos/pull/6416/files#diff-078a862b7f24a708ab9cb5323ad2768985ed7a68dd2e32540273717a05ecf723R23-R123

Reply:
- The lazy val should work as a memoized future, that looks good to me and seems to work well :+1: 
- I don’t think you’ll need the ApplicationLifecycle for this.
- Play will sometimes instantiate multiple instances of injected classes, and garbage-collect them again later. To ensure that this function is never run superfluously, you could add ElasticSearchClient to the `WebKnossosModule` setup, and bind it as singleton there. This is already done for a couple of other stateful services to enable shared caching. Restart after incremental compilation will still re-run everything in them.
- The `akka.pattern.after` seems like a nice way to sleep without blocking the thread from the thread pool. Didn’t know about this :)

Reply:
Thanks for looking into this!